lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -901,6 +901,26 @@ SPIRVFunction *LLVMToSPIRVBase::transFunctionDecl(Function *F) {
 
   transFPGAFunctionMetadata(BF, F);
 
+  if (auto *RegisterAllocModeMD = F->getMetadata("RegisterAllocMode")) {
```

Comment:
Probably should move this code into a separate function like:
`transFunctionMetadataAsUserSemanticDecoration` to separate this a bit

Reply:
good idea, done in latest commit